code/game/objects/items/weapons/storage/holster.dm

```diff
@@ -0,0 +1,269 @@
+	icon_state = "belt_holster"
+	item_state = "belt_holster"
+	rarity_value = 40
+
+	slot_flags = SLOT_BELT
+	price_tag = 240
+	matter = list(MATERIAL_BIOMATTER = 5)
+
+	sound_in = 'sound/effects/holsterin.ogg'
+	sound_out = 'sound/effects/holsterout.ogg'
+
+	w_class = ITEM_SIZE_NORMAL
+	max_w_class = ITEM_SIZE_HUGE
+
+	storage_slots = 2
+
+//Sheath
+/obj/item/storage/pouch/holster/belt/sheath
+	name = "sheath"
+	desc = "A sturdy brown leather sheath with a gold trim, made to house a variety of swords."
+	icon = 'icons/obj/sheath.dmi'
+	icon_state = "sheath_0"
+	item_state = "sheath_0"
+
+	price_tag = 120
+
+	storage_slots = 1
+
+	sound_in = 'sound/effects/sheathin.ogg'
+	sound_out = 'sound/effects/sheathout.ogg'
+
+	can_hold = list(
+		/obj/item/tool/sword/nt,
+		/obj/item/tool/sword/nt_sword,
+		/obj/item/tool/sword/saber,
+		/obj/item/tool/sword/katana,
+		/obj/item/tool/sword/katana/nano,
+		/obj/item/tool/sword,
+		/obj/item/toy/katana,
+		/obj/item/tool/sword/improvised
+		)
+	cant_hold = list(
+		/obj/item/tool/knife/dagger/nt,
+		/obj/item/tool/sword/nt/halberd,
+		/obj/item/tool/sword/nt/spear
+		)
+
+/obj/item/storage/pouch/holster/belt/sheath/improvised
+	name = "makeshift sheath"
+	desc = "A sturdy metal sheath with a rough finish. There's killing to do, draw your junkblade."
+	icon_state = "sheath_scrapsword"
+	price_tag = 50
+	matter = list(MATERIAL_STEEL = 2, MATERIAL_PLASTIC = 1)
+	can_hold = list(/obj/item/tool/sword/improvised)
+
+//Accessory holsters
+/obj/item/clothing/accessory/holster
+	name = "concealed carry holster"
+	desc = "An inconspicious holster that can be attached to your uniform. Can only fit small handguns and knives... Maybe something else, too."
+	icon_state = "concealed_carry"
+	slot = "utility"
+	matter = list(MATERIAL_BIOMATTER = 5)
+	price_tag = 160
+	spawn_blacklisted = FALSE
+	spawn_tags = SPAWN_TAG_HOLSTER
+	var/obj/item/storage/internal/holster
+	var/list/can_hold = list(
+		/obj/item/gun/projectile/giskard,
+		/obj/item/gun/projectile/selfload,
+		/obj/item/gun/energy/gun/martin,
+		/obj/item/tool/knife,
+		/obj/item/reagent_containers/food/snacks/mushroompizzaslice,
+		/obj/item/reagent_containers/food/snacks/meatpizzaslice,
+		/obj/item/reagent_containers/food/snacks/vegetablepizzaslice,
+		/obj/item/bananapeel
+		)
+
+/obj/item/clothing/accessory/holster/attack_hand(mob/user as mob)
+	. = ..()
+
+/obj/item/clothing/accessory/holster/attackby(obj/item/W as obj, mob/user as mob)
+	. = ..()
+
+/obj/item/clothing/accessory/holster/on_attached(obj/item/clothing/under/S, mob/user as mob)
+	..()
+
+	holster = new /obj/item/storage/internal(S)
+	holster.storage_slots = 1
+	holster.can_hold = can_hold
+	holster.max_w_class = ITEM_SIZE_SMALL
+	holster.master_item = S
+
+/obj/item/clothing/accessory/holster/on_removed(mob/user as mob)
+	QDEL_NULL(holster)
+	..()
+
+//For the holster hotkey
+//This verb is universal to any subtype of pouch/holster.
+/obj/item/storage/pouch/holster/verb/holster_verb()
```

Comment:
Might as well get rid of verb (they're bad for performance or so I've been told, m'okay) and use proc instead.

Reply:
Only verbs can be used by hotkeys. This is why this has to be a verb.

Reply:
Touché, that's certified BYOND moment.